### DISCUSSION TITLE
I have a volume mesh, how would I extract a surface mesh?

### DISCUSSION THREAD

**Discussion by tpat95** [SCIInstitute/SCIRun#2186] [Q&A]
I have used a cleaver to generate a volumetric mesh with air, heart, and blood pool components with (1), (2), and (3) dictating which elements correspond to what anatomy.

**Comment by nids2001** [comment]
We want a surface mesh WITH blood pools:

- In SCIRun load in the geometry using ReadField. 
- Now we want to isolate only the portions of the mesh that we want to extract the surfaces of.(If we want blood pools to be part of the surface, then we need to extract the myocardium alone. If we only want the epicardial surface than we extract the surface of the combined mesh elements.) Using the ClipFieldbyFunction module with the clipping location option set to 'Element Center' set the function to be `DATA1 == 1;` The output of this will be only the mesh elements belonging to the 1 designation. If this module fails, make sure you set the clipping location. 
- Now we want to extract the surface mesh. Using the GetFieldBoundary module (simply pass the output from the ClipFieldbyFunction into the first Port) SCIRun will extract the surface mesh. 
- Now we want to save the surface mesh. Using the WriteField module save you file where you want and with whatever name with the correct file type. 
- Done!

Your network should look like [ReadField -\> ClipFieldbyFunction(element center) -\> GetFieldBoundary -\> WriteField]
